### PR TITLE
[stdlib] Set, Dictionary: Fix (cocoa, native) equatable implementation

### PR DESCRIPTION
**Pull request by AndrewSB**
I was looking through the implementation of set and this change felt off to me.

@lorentey it looks like you changed this last, what do you think?

### CODE REVIEW

File: stdlib/public/core/Set.swift [hunk 1]
```diff
@@ -415,7 +415,7 @@ extension Set: Equatable {
     case (true, false):
       return lhs._variant.asNative.isEqual(to: rhs._variant.asCocoa)
     case (false, true):
-      return rhs._variant.asNative.isEqual(to: lhs._variant.asCocoa)
```
Comment (lorentey) [inline stdlib/public/core/Set.swift:418]: I was knee deep in Set guts trying to figure out how this managed to remain undetected for so long, but it turns out the original code is correct! Unlike the rest of the cases, here `isEqual(to:)` is called on `rhs`, not `lhs`. (`_CocoaSet.isEqual(to:)` does not support a native argument, so this line wouldn't compile with the suggested change.)

This is way too subtle. I blame the terrible naming convention `lhs` and `rhs` -- naming these parameters `left` and `right` would've highlighted the swap much better. Would you like to rename them?

**Review by lorentey — CHANGES_REQUESTED** [review]

File: stdlib/public/core/Set.swift (hunk 1 above)
Comment (AndrewSB) [reply stdlib/public/core/Set.swift:418]: oh! I see, in this case the `lhs` and `rhs`were reversed! The naming does make it hard to notice that. Will rename, sorry for the false alarm

**Comment by AndrewSB** [comment]
@lorentey I think this is ready for another pass & potentially to be merged

**Comment by lorentey** [comment]
@AndrewSB The current version won't work, either, for the same reason I wrote above:

> `_CocoaSet.isEqual(to:)` does not support a native argument, so this line wouldn't compile with the suggested change.

Stepping back, what concrete problem are you addressing with this PR? Have you verified that your change builds & runs locally before submitting this for review?

**Comment by lorentey** [comment]
@swift-ci smoke test

**Comment by lorentey** [comment]
```
13:34:51 
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/stdlib/public/core/Dictionary.swift:1585:60: error: cannot convert value of type '_NativeDictionary<Key, Value>' to expected argument type '__CocoaDictionary'
13:34:51       return lhs._variant.asCocoa.isEqual(to: rhs._variant.asNative)
13:34:51                                                            ^
13:34:51 /Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/swift/stdlib/public/core/Set.swift:418:60: error: cannot convert value of type '_NativeSet<Element>' to expected argument type '__CocoaSet'
13:34:51       return lhs._variant.asCocoa.isEqual(to: rhs._variant.asNative)
13:34:51                                                            ^
```

**Comment by gribozavr** [comment]
Assuming there is a bug in this code that this PR fixes, please add tests for it -- thanks!

**Comment by lorentey** [comment]
@AndrewSB I'm closing this as it isn't building in this stage, and I don't see what problem these changes are addressing. Feel free to submit a new PR if there is an issue to be resolved.

**Pull request closed without merging**